### PR TITLE
[SDSP-2065] CI pipeline improvements

### PR DESCRIPTION
**Pull request by st0ck**
Added
- ruby files linter
- ruby tests

Rubocop config defaults are still to be decided.

Jira ticket: https://adgear.atlassian.net/browse/SDSP-2065

[SDSP-2065]: https://adgear.atlassian.net/browse/SDSP-2065?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,6 +11,7 @@ spec/reports
 test/tmp
 test/version_tmp
 tmp
+Gemfile.lock
```
Comment (ittsel-ali) [inline .gitignore:15]: Hey Pavel, was there a reason to remove lock file? Gems normally comes with lock file to keep version control on gem dependencies

Comment (st0ck) [reply .gitignore:15]: Nice catch! Looks like my knowledge about it is outdated. I thought it should be gitignored for gems.
Just for the reference proving I wasn't right https://bundler.io/v2.4/man/bundle-install.1.html#THE-GEMFILE-LOCK

Thank you Ittsel! I'll return all `Gemfile.lock`s back.

File: Gemfile.lock [hunk 2]
```diff
@@ -1,15 +1,37 @@
...
+      rainbow (>= 2.2.2, < 4.0)
+      regexp_parser (>= 1.8, < 3.0)
+      rexml (>= 3.2.5, < 4.0)
+      rubocop-ast (>= 1.23.0, < 2.0)
+      ruby-progressbar (~> 1.7)
+      unicode-display_width (>= 1.4.0, < 3.0)
+    rubocop-ast (1.24.1)
+      parser (>= 3.1.1.0)
+    ruby-progressbar (1.11.0)
+    unicode-display_width (2.4.2)
```
Comment (ittsel-ali) [inline Gemfile.lock:34]: Hey Pavel, thanks for resolving the previous concern. I have one more question: Has this new update been tested by bundling the gems on the rtb-trader app? does it works fine or have any conflicts there?

Comment (st0ck) [reply Gemfile.lock:34]: `Gemfile.lock` file of the gem is ignored when installing the gem in a rails application so it won't affect rtb-trader. `Gemfile.lock` is needed only for the convenience of the gem contributors.

**Review by ittsel-ali — APPROVED** [review]

**Pull request merged**